kerl

```diff
@@ -411,8 +411,10 @@ assert_perl()
 
 get_javac_version()
 {
-    java_loc=$(which javac)
-    if [ -z "$java_loc" ]; then
+    # http://apple.stackexchange.com/a/135079
+    /usr/libexec/java_home >/dev/null 2>&1
```

Comment:
Can we roll this java-related change into a separate PR? Then we can go ahead and merge the rest of this and figure out the java stuff without blocking it?